packages/frontend/src/components/EditorSettings/index.tsx

```diff
@@ -0,0 +1,94 @@
+import { ElementType, ReactNode, useMemo, useState } from 'react'
+import { BiMailSend } from 'react-icons/bi'
+import { Navigate, useParams } from 'react-router-dom'
+import {
+  Box,
+  Divider,
+  Flex,
+  useBreakpointValue,
+  VStack,
+} from '@chakra-ui/react'
+import * as URLS from 'config/urls'
+import useAuthentication from 'hooks/useAuthentication'
+
+import EditorDrawer from './EditorDrawer'
+import EditorSidebar from './EditorSidebar'
+import Navbar from './Navbar'
+
+export type DrawerLink = {
+  Icon: ElementType
+  text: string
+  to: string
+}
+
+export interface EditorSettingsLayoutProps {
+  children: ReactNode
+}
+
+export default function EditorSettingsLayout(
+  props: EditorSettingsLayoutProps,
+): React.ReactElement {
+  const { children } = props
+
+  const { currentUser } = useAuthentication()
+
+  const { flowId } = useParams()
+  const [isDrawerOpen, setDrawerOpen] = useState(false)
+
+  const openDrawer = () => setDrawerOpen(true)
+  const closeDrawer = () => setDrawerOpen(false)
+
+  const drawerLinks = useMemo(
+    () => [
+      {
+        Icon: BiMailSend,
+        text: 'Email notifications',
+        to: URLS.FLOW_EDITOR_NOTIFICATIONS(flowId),
+      },
+    ],
+    [flowId],
+  )
```

Comment:
nit: since you already using `useMemo`, can do this also

```suggestion
  const openDrawer = () => setDrawerOpen(true)
  const closeDrawer = () => setDrawerOpen(false)

  const [drawerLinks, openDrawer, closeDrawer] = useMemo(
    () => [
      [{
        Icon: BiMailSend,
        text: 'Email notifications',
        to: URLS.FLOW_EDITOR_NOTIFICATIONS(flowId),
      }],
      () => setDrawerOpen(true),
      () => setDrawerOpen(false),
    ],
    [flowId, setDrawerOpen],
  )
```

can also consider omitting `useMemo` here since `URLS.FLOW_EDITOR_NOTIFICATIONS` isn't really an expensive computation.